src/main/java/gov/usgs/earthquake/indexer/DefaultIndexerListener.java

```diff
@@ -133,10 +133,14 @@ public boolean accept(IndexerEvent change) throws Exception {
 				Iterator<Event> iter = events.iterator();
 				while (iter.hasNext()) {
 					Event event = iter.next();
-					ProductSummary preferred = event
-							.getPreferredProduct(productType);
-					if (preferred != null && preferred.getId().equals(
-							change.getSummary().getId())) {
+					ProductSummary preferred = event.getPreferredProduct(productType);
+					if (preferred == null) {
```

Comment:
@emartinez-usgs : Since this is potentially looking at multiple events, should this only work when there is one event? or check that this product is associated to this event?

Main concern is what if there are multiple events, but there is only a shakemap in one...

Reply:
I think the second commit addresses this potential edge case